apps/web/src/components/editor/index.tsx

```diff
@@ -624,9 +625,15 @@ function EditorChrome(props: PropsWithChildren<EditorProps>) {
           variant="columnFill"
           className="editor"
           sx={{
-            alignSelf: ["stretch", focusMode ? "center" : "stretch", "center"],
+            alignSelf: [
+              "stretch",
+              focusMode ? "center" : "stretch",
+              !editorMargins ? (isTOCVisible ? "stretch" : "center") : "center"
+            ],
             maxWidth: editorMargins ? "min(100%, 850px)" : "auto",
-            width: "100%"
+            width: isTOCVisible
+              ? `calc(100% - ${TABLE_OF_CONTENTS_WIDTH}px)`
+              : "100%"
```

Comment:
I don't think this is the best way to do this. Instead of altering the width of the editor, we can render the TableOfContents to the right of the editor. That way it'll automatically resize when TOC becomes visible.